src/components/author.svelte

```diff
@@ -1,30 +1,71 @@
 <script lang="ts">
+  import Icon from './icon.svelte';
   import LinkedinSquare from './icons/linkedin-square.svelte';
   import TwitterSquare from './icons/twitter-square.svelte';
   import Email2 from './icons/email2.svelte';
 
   import type { StaffMember } from 'src/types';
+
   export let author: StaffMember;
+
+  let iconOptions = [
+    {
+      type: 'linkedin',
+      hasProperty: author.linkedInURL
+    },
+    {
+      type: 'twitter',
+      hasProperty: author.twitterURL
+    },
+    {
+      type: 'email',
+      hasProperty: author.hasEmail
+    }
+  ];
 </script>
 
+<br />
 <div class="author-container">
   <div class="bio-container">
-    <img src={author.imgURL} alt={author.name} class="author-image" />
+    <img src={author.imgURL} alt={author.firstName} class="author-image" />
 
     <div class="bio">
-      <h2>{author.name}</h2>
-      <p class="base-text">{author.miniBio}</p>
+      <h3>{author.firstName} {author.lastName}</h3>
+      <p>{author.miniBio}</p>
     </div>
   </div>
   <div class="cta-container">
-    <b>{author.projectStatus}</b>
+    {#if author.isAcceptingProjects === true}
+      <p class="bold-text teal">{author.firstName} is accepting new projects!</p>
+    {:else}
+      <p class="bold-text red">{author.firstName} isn't currently accepting new projects.</p>
+    {/if}
     <div class="social-container">
-      <p class="base-text">Contact {author.name}:</p>
-      <LinkedinSquare size="1.25rem" fillColor="#0b2647" hoverColor="#ffce5c" />
-      {#if author.twitterURL}
-        <TwitterSquare size="1.25rem" fillColor="#0b2647" hoverColor="#ffce5c" />
-      {/if}
-      <Email2 size="1.25rem" fillColor="#0b2647" hoverColor="#ffce5c" />
+      <p class="bold-text">Contact {author.firstName}:</p>
+
+      {#each iconOptions as option}
+        {#if option.hasProperty}
+          {#if option.type === 'linkedin'}
+            <a href={author.linkedInURL} target="_blank" rel="noopener noreferrer">
+              <Icon>
+                <LinkedinSquare />
```

Comment:
instead of doing `if option.type = _____`, why not put the component reference on the hashmap itself? it'll save you ~25 lines of repetitive code

Reply:
I tried a couple different ways of doing that initially, but it wasn't working. I'll do some research on the correct syntax and try again.

Reply:
Got it working! :)